### PR TITLE
lowers megafauna chance, forces a bdm spawn on lavaland

### LINKED ISSUE #69677: too many colossus on icebox
Round ID: [189747](https://scrubby.melonmesa.com/round/189747)
BYOND Version: 514.1583
Server revision compiled on: 
Master commit: [2e29dd1989f81a79c032aa16cf870be49bd220a0](https://www.github.com/tgstation/tgstation/commit/2e29dd1989f81a79c032aa16cf870be49bd220a0)
The following pull requests are currently test merged:
[#69634: 'Moves "catch this var/flag" code from obj/init and datum/new into the types that use it ' by LemonInTheDark at commit 18b5e81303](https://www.github.com/tgstation/tgstation/pull/69634)
[#69632: 'Macro optimizes SSmapping saving 50% ' by LemonInTheDark at commit 90f63bbbeb](https://www.github.com/tgstation/tgstation/pull/69632)
[#69240: 'Changed Supermatter Internal Math + UI Additions' by vincentiusvin at commit 09a016a732](https://www.github.com/tgstation/tgstation/pull/69240)

Local commit: fb5219713dcd39d4fe8afda34a76c0a0ebba5c63
TGS version: /tg/station 13 Server v3.2.6.2
DMAPI version: 3.2.1.3

Current Informational Settings:
Protect Authority Roles From Traitor: 1
Protect Assistant Role From Traitor: 0
Enforce Human Authority: 1
Allow Latejoin Antagonists: 1

observe on icebox and see way too many colossus.
image

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request
quarters the chance of megafauna spawns on icebox (this only applies to the colossi)
halves the chance of megafauna spawns on lavaland
forces a bdm spawn on lavaland
closes #69677

## Why It's Good For The Game
26 colossi is not fun
image

fighting megafauna is fun and ruin rng screwing you out of one is not fun

## Changelog
:cl:
balance: less megafauna should spawn on icebox and lavaland
qol: blood drunk miner ruin on lavaland will spawn every round
/:cl:

### CODE REVIEW

File: code/datums/ruins/lavaland.dm [hunk 1]
```diff
@@ -150,6 +150,15 @@
...
 
+/datum/map_template/ruin/lavaland/blood_drunk_miner/random
+	name = "Blood-Drunk Miner (Random)"
+	suffix = null
+	always_place = TRUE
+
+/datum/map_template/ruin/lavaland/blood_drunk_miner/random/New()
+	suffix = pick("lavaland_surface_blooddrunk1.dmm", "lavaland_surface_blooddrunk2.dmm", "lavaland_surface_blooddrunk3.dmm")
+	return ..()
+
```
Comment (msgerbs) [inline code/datums/ruins/lavaland.dm:161]: Shouldn't you then remove the existing BDM ruins from this file?

Comment (Fikou) [reply code/datums/ruins/lavaland.dm:161]: no, admins may want to spawn them.

Comment (msgerbs) [reply code/datums/ruins/lavaland.dm:161]: Makes sense

**Comment by itsmeow** [comment]
> what caused the sudden explosion in both megafauna and tendril/portal spawning in the first place?

It appears to be a bug in https://github.com/tgstation/tgstation/pull/69632

This PR does not actually fix the underlying issue from what I understand

**Comment by itsmeow** [comment]
Looking into it a bit more, the optimization is sort of necessary, and a better solution for spacing mob spawns out would be ideal, but this does seem to sufficiently patch the issue.

Here's what I ended up doing https://github.com/BeeStation/BeeStation-Hornet/pull/9469/commits/cfac27d724d0091e7c7eb21322e49caadca05278

**Pull request merged**